### ISSUE TITLE
The user is not redirected back to AMO after reset password 

### ISSUE THREAD

**Issue by ioanarusiczki** [mozilla/fxa#15902]
I could reproduce this from AMO dev - fxa stage and AMO stage - fxa prod as follows: 

1. Login from addons.allizom.org
2. Click on forgot password and begin reset
3. After getting the email and resetting go back in the tab where you begin the reset and try to login 

Expected result:
I'll be redirected to AMO

Actual result:
I'm landing on the fxA page (there's no autenticate-callback in the network)

I could reproduce this with a new account (without 2fa) but also with an account with 2fa enabled.
I'll attach a gif (it's a bit longer since I tried to catch the entire process)

![fxA page displayed not amo](https://github.com/mozilla/fxa/assets/33448286/bf3ec805-7fdb-4b70-b210-451eee73142d)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXA-8460)

**Comment by ioanarusiczki** [comment]
This is still reproducible with AMO stage.
When resetting the page displayed is the one below. To login to AMO it requires a new tab.

![still reproducible](https://github.com/user-attachments/assets/dde47071-c1ae-4d8d-9b1f-d7a901cda485)

**Comment by vpomerleau** [comment]
This issue has been resolved and it is now possible to continue to AMO after password reset. 

See for stage:

![Image](https://github.com/user-attachments/assets/cf9f769b-1095-4144-81af-24bf9e5f998c)

**Comment by ioanarusiczki** [comment]
Yeah, I've checked with AMO dev and stage this and now I'm redirected to AMO - sometimes on homepage but I've also landed on edit profile page , but as long as it's AMO it should be good.

this would be on production (for fxa) and landing on AMO's homepage

![Image](https://github.com/user-attachments/assets/cd11b200-8084-4a84-8aa2-9b100743486f)

this is with stage and landing on user's edit profile page

![Image](https://github.com/user-attachments/assets/46d56495-3744-4894-ba59-266943f60828)
